source/features/show-whitespaces.tsx

```diff
@@ -0,0 +1,96 @@
+import './show-whitespaces.css';
+import React from 'dom-chef';
+import select from 'select-dom';
+import features from '../libs/features';
+import onPrFileLoad from '../libs/on-pr-file-load';
+import onNewComments from '../libs/on-new-comments';
+import getTextNodes from '../libs/get-text-nodes';
+
+function showWhiteSpacesOn(line: Element): void {
+	const textNodes = getTextNodes(line);
+
+	for (const textNode of textNodes) {
+		const nodeValue = textNode.textContent!;
+		if (nodeValue.length === 0) {
+			continue;
+		}
+
+		const fragment = document.createDocumentFragment();
+
+		let lastEncounteredCharType;
+		let charType: 'space' | 'tab' | 'other';
+		let node;
+
+		for (const char of nodeValue) {
+			if (char === ' ') {
+				charType = 'space';
+			} else if (char === '\t') {
+				charType = 'tab';
+			} else {
+				charType = 'other';
+			}
+
+			if ((lastEncounteredCharType && lastEncounteredCharType === charType) && node) {
```

Comment:
As a consequence of the previous suggestion, I think you can now shorten this to:

```suggestion
			if (node && lastEncounteredCharType === charType) {
```